File: Original/notification_service.py
"""
notification_service.py - NotificationService component (Baseline).
Corresponds to the NotificationService lifeline in the sequence diagram.
Responsibility: send outcome-specific notifications.
NOTE: Three separate methods for acceptance/rejection/revision — an identified
      inefficiency (scattered conditional notification logic).
"""
from .metrics import tracker


class NotificationService:
    """Dispatches outcome notifications.

    Sequence diagram interactions (ALT block):
      - EvaluationManager -> notifyAcceptance() -> NotificationService  [if accepted]
      - EvaluationManager -> notifyRejection()  -> NotificationService  [if rejected]
      - EvaluationManager -> notifyRevision()   -> NotificationService  [if revision]
    """

    @tracker.track
    def notifyAcceptance(self, submission_id: str) -> dict:
        return {
            "submission_id": submission_id,
            "message": f"Submission {submission_id} has been ACCEPTED.",
            "type": "acceptance",
        }

    @tracker.track
    def notifyRejection(self, submission_id: str) -> dict:
        return {
            "submission_id": submission_id,
            "message": f"Submission {submission_id} has been REJECTED.",
            "type": "rejection",
        }

    @tracker.track
    def notifyRevision(self, submission_id: str) -> dict:
        return {
            "submission_id": submission_id,
            "message": f"Submission {submission_id} requires REVISION.",
            "type": "revision",
        }

File: Original/src/NotificationService.java
public class NotificationService {

    private UI ui;

    public NotificationService(UI ui) {
        this.ui = ui;
    }

    public void notifyAcceptance() {
        MetricTracker.record("NotificationService.notifyAcceptance");
        ui.sendNotification("Your submission has been accepted.");
    }

    public void notifyRejection() {
        MetricTracker.record("NotificationService.notifyRejection");
        ui.sendNotification("Your submission has been rejected.");
    }

    public void notifyRevision() {
        MetricTracker.record("NotificationService.notifyRevision");
        ui.sendNotification("Your submission requires revision.");
    }
}
